File: typescript-client-sdk/src/sdk/models/shared/numberfilter.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class NumberFilter extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=equals" })
  equals?: number;

  @SpeakeasyMetadata({ data: "json, name=gt" })
  gt?: number;

  @SpeakeasyMetadata({ data: "json, name=gte" })
  gte?: number;

  @SpeakeasyMetadata({ data: "json, name=in" })
  in?: number[];

  @SpeakeasyMetadata({ data: "json, name=lt" })
  lt?: number;

  @SpeakeasyMetadata({ data: "json, name=lte" })
  lte?: number;

  @SpeakeasyMetadata({ data: "json, name=not" })
  not?: number;

  @SpeakeasyMetadata({ data: "json, name=notIn" })
  notIn?: number[];
}

File: go-client-sdk/pkg/models/shared/numberfilter.go
package shared

type NumberFilter struct {
	Equals *float64  `json:"equals,omitempty"`
	Gt     *float64  `json:"gt,omitempty"`
	Gte    *float64  `json:"gte,omitempty"`
	In     []float64 `json:"in,omitempty"`
	Lt     *float64  `json:"lt,omitempty"`
	Lte    *float64  `json:"lte,omitempty"`
	Not    *float64  `json:"not,omitempty"`
	NotIn  []float64 `json:"notIn,omitempty"`
}
